test/performance/alignment/global_affine_alignment_simd_benchmark.cpp

```diff
@@ -0,0 +1,185 @@
+#include <seqan3/alphabet/aminoacid/aa20.hpp>
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/range/views/to.hpp>
+#include <seqan3/range/views/zip.hpp>
+#include <seqan3/std/ranges>
+#include <seqan3/test/performance/units.hpp>
+#include <seqan3/test/seqan2.hpp>
+
+#include <seqan3/core/debug_stream.hpp>
+
+#ifdef SEQAN3_HAS_SEQAN2
+    #include <seqan/align.h>
+    #include <seqan/align_parallel.h>
+#endif
+
+#include "test_sequence_generation.hpp" // Shared functionality to generate test sequences.
+
+using namespace seqan3;
+using namespace seqan3::test;
+
+constexpr auto affine_cfg = align_cfg::mode{global_alignment} |
+                            align_cfg::gap{gap_scheme{gap_score{-1}, gap_open_score{-10}}} |
+                            align_cfg::scoring{nucleotide_scoring_scheme{match_score{4}, mismatch_score{-5}}};
+
+// Aliases to beautify the benchmark output
+using score = detail::with_score_type;
+using end_position = detail::with_back_coordinate_type;
+using trace = detail::with_alignment_type;
+
+// Globally defined constants to ensure same test data.
+inline constexpr size_t sequence_length_mean = 150;
+inline constexpr size_t set_size        = 1024;
+
+// Range to test for sequence length variance
+inline constexpr size_t deviation_begin = 0;
+inline constexpr size_t deviation_end = 64;
+inline constexpr size_t deviation_step = 8;
+
+// ============================================================================
+//  affine; score; dna4; collection
+// ============================================================================
+
+template <typename result_t>
+void seqan3_affine_dna4_simd(benchmark::State & state)
+{
+    size_t sequence_length_deviation = state.range(0);
+    auto [vec1, vec2] = generate_data_seqan3<seqan3::dna4>(sequence_length_mean, sequence_length_deviation, set_size);
+
+    auto data = views::zip(vec1, vec2) | views::to<std::vector>;
+
+    int64_t total = 0;
+    for (auto _ : state)
+    {
+        for (auto && res : align_pairwise(data, affine_cfg |
+                                          align_cfg::result{result_t{}, using_score_type<int16_t>} |
+                                          align_cfg::vectorise))
+        {
+            total += res.score();
+        }
+    }
+
+    state.counters["cells"] = pairwise_cell_updates(views::zip(vec1, vec2), affine_cfg);
+    state.counters["CUPS"] = cell_updates_per_second(state.counters["cells"]);
+    state.counters["total"] = total;
+}
+
+BENCHMARK_TEMPLATE(seqan3_affine_dna4_simd, score)
+                        ->UseRealTime()
+                        ->DenseRange(deviation_begin, deviation_end, deviation_step);
+BENCHMARK_TEMPLATE(seqan3_affine_dna4_simd, end_position)
+                        ->UseRealTime()
+                        ->DenseRange(deviation_begin, deviation_end, deviation_step);
+// TODO: enable me BENCHMARK_TEMPLATE(seqan3_affine_dna4_simd, trace)->UseRealTime();
```

Comment:
open todo

Reply:
Yes, but not yet part of this PR. Because first I need the vectorised alignment with trace. And there is a PR, but it will change a little so that will be tackled later. If you insist, then I would remove the comment otherwise remove it when it is added.

Reply:
Well, I general I'm against todos as comments and in favour of cards (or another tracking), since they might still be forgotten and "pollute" the code. I'm always reminded  of all the comments in seqan2 :see_no_evil: